src/wordlist/wordlist.ts

```diff
@@ -0,0 +1,95 @@
+import {RBTree} from 'bintrees';
+
+import words from "@/wordlist/words.json";
+
+type WordTree = RBTree<string>;
```

Comment:
Using a binary search tree like Red Black tree seem like a great idea! It should ensure efficient results for insertions (one time at the beginning, but still ..) and search..

Reply:
Thanks! My concern was that we'll be searching for match quite frequently (every time the input changes).